src/drivers/dw-dma.c

```diff
@@ -465,11 +466,15 @@ static int dw_dma_start(struct dma *dma, int channel)
 	dw_write(dma, DW_CFG_LOW(channel), chan->cfg_lo);
 	dw_write(dma, DW_CFG_HIGH(channel), chan->cfg_hi);
 
-	if (chan->timer_delay)
+	if (chan->timer_delay) {
+		/* add offset for capture to handle external interface start */
+		if (chan->direction == DMA_DIR_DEV_TO_MEM)
+			start_offset = PLATFORM_TIMER_START_OFFSET;
+
 		/* activate timer for timer driven scheduling */
 		work_schedule_default(&chan->dma_ch_work,
-				      chan->timer_delay);
-	else if (chan->status == COMP_STATE_PREPARE)
+				      chan->timer_delay + start_offset);
+	} else if (p->chan[channel].status == COMP_STATE_PREPARE)
```

Comment:
why has "chan->" been replaced with "p->chan[channel]."? @tlauda could you push an incremental patch to fix that?

Reply:
Good catch, I think @tlauda wrote this prior to the cleanup work. Incremental PR is fine.